File: Backend/Models/Entities/Procurement_Request.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ProcionAPI.Models.Entities
{
    public class Procurement_Request
    {
        [Key]
        public int Procurement_Request_ID { get; set; }
        [Required]
        public int Vendor_ID { get; set; }
        [ForeignKey("Vendor_ID")]
        public Vendor Vendor { get; set; }
        [Required]
        public int Requisition_Status_ID { get; set; }
        [ForeignKey("Requisition_Status_ID")]
        public Requisition_Status Requisition_Status { get; set; }
        [Required]
        public int User_ID { get; set; }
        [ForeignKey("User_ID")]
        public User User { get; set; }
        [Required]
        public string Name { get; set; }
        [Required]
        public string Description { get; set; }



    }
}

File: IonicFrontend/Procionic/src/app/Shared/Procurement_Request.ts
import { Requisition_Status } from "./Requisition_Status";
import { User } from "./User";
import { VendorOnboardRequest } from "./VendorOnboardRequest";


export interface Procurement_Request {
    procurement_Request_ID: Number;
    vendor_ID: Number;
    vendor: VendorOnboardRequest;
    requisition_Status_ID: Number;
    requisition_Status: Requisition_Status;
    user_ID: Number;
    user: User;
    name: String;
    description: String;
}